_posts/2023-11-30-prevent-broker-scale-down-if-containing-paritition-replicas.md

```diff
@@ -0,0 +1,179 @@
+---
+layout: post
+title: "Preventing Kafka broker scale down if partition replicas are present"
+date: 2023-12-11
+author: Shubham Rawat
+---
+
+Apache Kafka is a platform designed for scalability.
+You can always scale in or scale out your Kafka Clusters based on your use case.
+When scaling down a cluster, it's crucial to ensure that data across brokers is moved or copied throughout the cluster.
+You may experience data loss if you forget to move out partition replicas from the broker being removed.
```

Comment:
Would it be also worth mentioning that Strimzi might not be able to roll such cluster? I think that is more common than a data loss (which can also happen in some cases, so no need to remove it).